dask/array/tests/test_reductions.py

```diff
@@ -121,32 +128,44 @@ def reduction_2d_test(da_func, darr, np_func, narr, use_dtype=True):
         assert eq(da_func(darr, dtype='f8'), np_func(narr, dtype='f8'))
         assert eq(da_func(darr, dtype='i8'), np_func(narr, dtype='i8'))
 
+    if max_leaves:
+        assert eq(da_func(darr, max_leaves=max_leaves), np_func(narr))
+        assert eq(da_func(darr, keepdims=True, max_leaves=max_leaves),
+                  np_func(narr, keepdims=True))
+        assert eq(da_func(darr, axis=0, max_leaves=max_leaves), np_func(narr, axis=0))
+        assert eq(da_func(darr, axis=0, keepdims=True, max_leaves=max_leaves),
+                  np_func(narr, axis=0, keepdims=True))
+        assert eq(da_func(darr, axis=1, max_leaves=max_leaves), np_func(narr, axis=1))
+        assert eq(da_func(darr, axis=1, keepdims=True, max_leaves=max_leaves),
+                  np_func(narr, axis=1, keepdims=True))
+
 
 def test_reductions_2D_float():
     x = np.arange(1, 122).reshape((11, 11)).astype('f4')
     a = da.from_array(x, chunks=(4, 4))
+    max_leaves = {0: 2, 1: 2}
```

Comment:
I know you haven't really filled out tests, so please excuse pedantic comments, but should also verify the accuracy of the branching as well for a few different kinds of branching.